Tests/AMPDatabaseHelperTests.m

```diff
@@ -444,15 +444,45 @@ - (void)testUpgradeFromVersion1ToVersion3 {
 }
 
 - (void)testUpgradeFromVersion3ToVersion3{
-    // upgrade does nothing, can insert into event, store, long_store, identify
     [self.databaseHelper dropTables];
+    XCTAssertTrue([self.databaseHelper upgrade:0 newVersion:3]);
+    XCTAssertTrue([self.databaseHelper addEvent:@"test"]);
+    XCTAssertTrue([self.databaseHelper insertOrReplaceKeyValue:@"key" value:@"value"]);
+    XCTAssertTrue([self.databaseHelper insertOrReplaceKeyLongValue:@"key" value:[NSNumber numberWithLongLong:0LL]]);
+    XCTAssertTrue([self.databaseHelper addIdentify:@"test"]);
+
+    // upgrade does nothing, can insert into event, store, long_store, identify
+    // FIXME: Update from 3 to 3 drops all tables and recreates, seems like a bug.
+    // FIXME: Ask Daniel Jih.
```

Comment:
@djih looking at the `upgrade` logic, in the case of upgrading from latest version => latest version it will drop all tables and recreate from scratch. This allowed `testUpgradeFromVersion3ToVersion3` to pass previously, but I had to update the test to recreate tables for versions 0-3 after adding version 4.

Is this expected behavior? I realize it's been a while, last change was 2015 :)